spec/overcommit/hook/pre_commit/licence_header_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+
+describe Overcommit::Hook::PreCommit::LicenceHeader do
+  let(:config)  { Overcommit::ConfigurationLoader.default_configuration }
+  let(:context) { double('context') }
+  subject { described_class.new(config, context) }
+
+  context 'when licence file is missing' do
+    before do
+      subject.stub(:licence_lines).and_raise(Errno::ENOENT)
+    end
+
+    it { should fail_hook }
+  end
+
+  context 'with headers in all applicable files' do
+    let(:licence) do
+      ['Hello World', 'Copyleft 2017']
+    end
+
+    before do
+      subject.stub(:applicable_files).and_return(%w[main.go])
+      subject.stub(:licence_lines).and_return(licence)
+      subject.stub(:check_file).and_return(nil)
+    end
+
+    it { should pass }
+  end
+end
```

Comment:
I'd really like an "integration" spec where I lay out a repo with a LICENCE.txt file and a few files with and without licence headers. Looks like most of the tests here are stubs/doubles/mocks. Perhaps I've just been writing Go for too long, but am I missing an easier way in Ruby/rspec to run this closer to how it's actually going to run? I uncovered a ton of bugs in manual testing due to the amount of mock/expect/stub/and_return etc.